proxy/src/fs_watch.rs

```diff
@@ -122,14 +115,37 @@ impl PathAndHash {
     }
 
     fn has_changed(&self) -> io::Result<bool> {
-        let contents = fs::read(&self.path)?;
-        let hash = Some(digest::digest(&digest::SHA256, &contents[..]));
-        let changed = self.last_hash.borrow().as_ref().map(Digest::as_ref)
-            != hash.as_ref().map(Digest::as_ref);
-        if changed {
-            self.last_hash.replace(hash);
+        match fs::read(&self.path) {
+            Ok(contents) => {
+                // If we were able to read the file, compare the hash of its
+                // current contents with the previous hash to determine if it
+                // has changed.
+                let hash = Some(digest::digest(&digest::SHA256, &contents[..]));
+                let changed = self.last_hash.borrow().as_ref()
+                    .map(Digest::as_ref) != hash.as_ref().map(Digest::as_ref);
```

Comment:
this is pretty hard to parse. can you move the map up onto the prior line?

Reply:
finding a way of structuring this that was readable but also placated the borrow checker was kind of tricky; I'll take another crack at it though.